java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/handlers/CreateCategoryHandler.java

```diff
@@ -19,28 +19,27 @@
 @Component
 public class CreateCategoryHandler extends MenuBaseCommandHandler<CreateCategoryCommand> {
 
-  private UUID categoryId;
-
   public CreateCategoryHandler(
       MenuRepository menuRepository, ApplicationEventPublisher applicationEventPublisher) {
     super(menuRepository, applicationEventPublisher);
   }
 
   @Override
   Optional<UUID> handleCommand(Menu menu, CreateCategoryCommand command) {
-    menu.setCategories(addCategory(menu, command));
+    UUID categoryId = UUID.randomUUID();
+    menu.setCategories(addCategory(menu, command, categoryId));
     menuRepository.save(menu);
+    command.setCategoryId(categoryId);
```

Comment:
We can re-write this to be 
command.setCategoryId( UUID.randomUUID());
 menu.setCategories(addCategory(menu, command));
 menuRepository.save(menu);